### ISSUE TITLE
controlMode always returns control::Auto

### ISSUE THREAD

**Issue by HemilTheRebel** [agauniyal/rang#106]
controlMode always returns control::Auto. It does not make sense to do this in operator <<:
`const control option = rang_implementation::controlMode();
       switch (option) {
            case control::Auto:
                return rang_implementation::supportsColor()
                    && rang_implementation::isTerminal(os.rdbuf())
                  ? rang_implementation::setColor(os, value)
                  : os;
            case control::Force: return rang_implementation::setColor(os, value);
            default:    return os;
        }`

**Comment by agauniyal** [comment]
I feel I don't understand the problem, could you rephrase it again or maybe provide some example 🤔
